reactive/kubernetes_master.py

```diff
@@ -1405,25 +1409,62 @@ def create_tokens_and_sign_auth_requests():
 @when("kube-api-endpoint.available")
 def push_service_data():
     """Send configuration to the load balancer, and close access to the
-    public interface"""
+    public interface.
+    """
     kube_api = endpoint_from_flag("kube-api-endpoint.available")
 
-    external_endpoints = kubernetes_master.get_external_lb_endpoints()
-    if external_endpoints:
-        addresses = [e[0] for e in external_endpoints]
+    endpoints = kubernetes_master.get_endpoints_from_config()
+    if endpoints:
+        addresses = [e[0] for e in endpoints]
         kube_api.configure(kubernetes_master.STANDARD_API_PORT, addresses, addresses)
     else:
-        # no external addresses configured, so rely on the interface layer
+        # no manually configured LBs, so rely on the interface layer
         # to use the ingress address for each relation
         kube_api.configure(kubernetes_master.STANDARD_API_PORT)
 
 
-@when("certificates.available", "kube-api-endpoint.available", "cni.available")
+@when("leadership.is_leader")
+@when_any(
+    "endpoint.loadbalancer-internal.available",
+    "endpoint.loadbalancer-external.available",
+)
+def request_load_balancers():
+    """Request LBs from the related provider(s)."""
+    for lb_type in ("internal", "external"):
+        lb_provider = endpoint_from_name("loadbalancer-" + lb_type)
+        if not lb_provider.is_available:
+            continue
+        req = lb_provider.get_request("api-server-" + lb_type)
+        req.protocol = req.protocols.tcp
+        api_port = kubernetes_master.STANDARD_API_PORT
+        req.port_mapping = {api_port: api_port}
```

Comment:
It might be worth considering mapping 443 to api_port since that's how it has been with kubeapi-load-balancer in the past.